### PR TITLE
Regional mean f1 + clamping/multipliers fix

### PR DESCRIPTION
**Pull request by cbutsko**
Adding computation of regional mean f1 and possibility to use it as a stopping metric.

### CODE REVIEW

File: src/worldcereal/train/finetuning_utils.py [hunk 1]
```diff
@@ -494,6 +494,68 @@ def _records_to_scalar_metrics(records: List[dict]) -> dict[str, float]:
...
+            )
+        except Exception as exc:  # noqa: BLE001
+            logger.warning(f"Failed computing macro F1 for region '{region}': {exc}")
+
+    if len(regional_f1s) < 2:
+        return None
+    return float(np.mean(regional_f1s))
+
+
+_CROP_FPR_POSITIVE_CLASS = "temporary_crops"
```
Comment (kvantricht) [inline src/worldcereal/train/finetuning_utils.py]: this is hardcoded while it's normally set at the finetune level?

File: src/worldcereal/train/finetuning_utils.py [hunk 2]
```diff
@@ -1036,7 +1100,7 @@ def forward(
         loss = torch.zeros(1, device=device, dtype=torch.float32)
 
         # Landcover pathway – only activate for samples explicitly routed to the LC
-        # head (tasks[idx] == landcover_task_name).  When the SeasonalTaskBatchSampler is
+        # head (tasks[idx] == landcover_task_name).  When the DualHeadBatchSampler is
```
Comment (kvantricht) [inline src/worldcereal/train/finetuning_utils.py]: don't get why this is renamed back to DualHeadBatchSampler ?

Comment (cbutsko) [reply src/worldcereal/train/finetuning_utils.py]: not sure how that slipped in... fixed here https://github.com/WorldCereal/worldcereal-classification/pull/736/changes/e9e5369c74336e03b5101420e845fc972d374cfb

File: src/worldcereal/train/finetuning_utils.py (hunk 1 above)
Comment (cbutsko) [reply src/worldcereal/train/finetuning_utils.py]: yeah, good point. It was just a practical decision to hardcode it here. it's monitoring-only, and there's a very specific type of commission I wanted to add - non_cropland predicted as temporary_crops. 
Looking deeper, though, we already have that concept in this very file. So I've rewired the FPR function to reuse the cropland definition set at finetuning time (`cropland_class_names`) instead of assuming one here https://github.com/WorldCereal/worldcereal-classification/pull/736/changes/e26c82b74a29cb4030f967d1603b03735aa9c76c

**Review by kvantricht — APPROVED** [review]
Ready for testing :-)
